src/serverMain/kotlin/com/studo/campusqr/endpoints/ReportData.kt

```diff
@@ -72,30 +94,44 @@ suspend fun AuthenticatedApplicationCall.returnReportData() {
       ?.joinToString(separator = "")
       ?.take(20)
 
-  respondObject(
-      ReportData(
-          impactedUsersCount = impactedUsersEmails.count(),
-          impactedUsersMailtoLink = "mailto:?bcc=" + impactedUsersEmails.joinToString(";"),
-          impactedUsersEmailsCsvData = impactedUsersEmails.joinToString("\n"),
-          reportedUserLocations = reportedUserCheckIns.map { checkIn ->
-            ReportData.UserLocation(
-                email = checkIn.email,
-                date = checkIn.date.toAustrianTime(yearAtBeginning = false),
-                locationName = locationMap.getValue(checkIn.locationId),
-                seat = checkIn.seat,
-            )
-          }.toTypedArray(),
-          reportedUserLocationsCsv = "sep=;\n" + reportedUserCheckIns.joinToString("\n") {
-            "${it.email};${it.date.toAustrianTime(yearAtBeginning = false)};${locationMap.getValue(it.locationId)}"
-          },
-          reportedUserLocationsCsvFileName = "${csvFilePrefix?.plus("-checkins") ?: "checkins"}.csv",
-          startDate = oldestDate.toAustrianTime("dd.MM.yyyy"),
-          endDate = now.toAustrianTime("dd.MM.yyyy"),
-          impactedUsersEmailsCsvFileName = "${csvFilePrefix?.plus("-emails") ?: "emails"}.csv"
-      )
+  return ReportData(
+      impactedUsersCount = impactedUsersEmails.count(),
+      impactedUsersEmails = impactedUsersEmails.toTypedArray(),
+      impactedUsersMailtoLink = "mailto:?bcc=" + impactedUsersEmails.joinToString(";"),
+      impactedUsersEmailsCsvData = impactedUsersEmails.joinToString("\n"),
+      reportedUserLocations = reportedUserCheckIns.map { checkIn ->
+        ReportData.UserLocation(
+            email = checkIn.email,
+            date = checkIn.date.toAustrianTime(yearAtBeginning = false),
+            locationName = locationMap.getValue(checkIn.locationId),
+            seat = checkIn.seat,
+        )
+      }.toTypedArray(),
+      reportedUserLocationsCsv = "sep=;\n" + reportedUserCheckIns.joinToString("\n") {
+        "${it.email};${it.date.toAustrianTime(yearAtBeginning = false)};${locationMap.getValue(it.locationId)}"
+      },
+      reportedUserLocationsCsvFileName = "${csvFilePrefix?.plus("-checkins") ?: "checkins"}.csv",
+      startDate = oldestDate.toAustrianTime("dd.MM.yyyy"),
+      endDate = Date().toAustrianTime("dd.MM.yyyy"),
+      impactedUsersEmailsCsvFileName = "${csvFilePrefix?.plus("-emails") ?: "emails"}.csv"
   )
 }
 
+suspend fun AuthenticatedApplicationCall.returnReportData() {
+  if (!user.isModerator) {
+    respondForbidden()
+    return
+  }
+
+  val params = receiveJsonStringMap()
+
+  val now = Date()
+  val emails = params.getValue("email").split(*emailSeparators).filter { it.isNotEmpty() }
+  val oldestDate = params["oldestDate"]?.toLong()?.let { Date(it) } ?: now.addDays(-14)
```

Comment:
what is the 14 hardcoding here?

Reply:
default `oldestDate` for the reporting view. (this is not new code)

Reply:
Ahh